lib/process-services-cloud/src/lib/process/process-filters/components/edit-process-filter-cloud.component.ts

```diff
@@ -253,6 +258,21 @@ export class EditProcessFilterCloudComponent implements OnInit, OnChanges, OnDes
             });
     }
 
+    onAppNameChange() {
+        const appNameController = this.editProcessFilterForm?.get('appName');
+        if(appNameController) {
+            appNameController.valueChanges.subscribe((res) => {
```

Comment:
you are creating a subscription every time something change.

Reply:
Not on every change, Here we are listening only to the appName change once it changed we reset current appVersions and processDefinitions to empty then let component to fetch version based on new appName

Reply:
handled Subscription